### PR TITLE
Weight Decay + additions in train.py,model.py

### PR DESCRIPTION
**Pull request by saswatpp**

### CODE REVIEW

File: d2l/train.py [hunk 1]
```diff
@@ -71,5 +79,4 @@ def train_ch5(net, train_iter, test_iter, criterion, num_epochs, batch_size, dev
                 n += y.shape[0]
 
         test_acc = evaluate_accuracy(test_iter, net, device) 
-        print('epoch %d, loss %.4f, train acc %.3f, test acc %.3f, time %.1f sec'
-            % (epoch + 1, train_l_sum/n, train_acc_sum/n, test_acc, time.time() - start))
+        print('epoch %d, loss %.4f, train acc %.3f, test acc %.3f, time %.1f sec'% (epoch + 1, train_l_sum/n, train_acc_sum/n, test_acc, time.time() - start))
```
Comment (AnirudhDagar) [inline d2l/train.py]: Don't make this a single line. See PEP-8 formatting style. 80 tokens at max in a line.

File: d2l/train.py [hunk 2]
```diff
@@ -43,8 +52,7 @@ def train_ch3(net, train_iter, test_iter, criterion, num_epochs, batch_size, lr=
             train_acc_sum += torch.sum((torch.argmax(y_hat, dim=1).type(torch.FloatTensor) == y).detach()).float()
             n += list(y.size())[0]
         test_acc = evaluate_accuracy(test_iter, net)
-        print('epoch %d, loss %.4f, train acc %.3f, test acc %.3f'
-              % (epoch + 1, train_l_sum / n, train_acc_sum / n, test_acc))
+        print('epoch %d, loss %.4f, train acc %.3f, test acc %.3f'% (epoch + 1, train_l_sum / n, train_acc_sum / n, test_acc))
```
Comment (AnirudhDagar) [inline d2l/train.py]: Don't make this a single line. See PEP-8 formatting style. 80 tokens at max in a line.
